packages/google_sign_in/android/src/main/java/io/flutter/plugins/googlesignin/GoogleSignInPlugin.java

```diff
@@ -439,4 +477,41 @@ private void finishWithError(String errorCode, String errorMessage) {
     }
     pendingOperation = null;
   }
+
+  private class GoogleApiClientConnectionManager implements ActivityLifecycleCallbacks {
+    @Override
+    public void onActivityCreated(Activity activity, Bundle bundle) {
+      resolvingError = bundle != null && bundle.getBoolean(STATE_RESOLVING_ERROR, false);
+    }
+
+    @Override
+    public void onActivityDestroyed(Activity activity) {}
+
+    @Override
+    public void onActivityPaused(Activity activity) {}
+
+    @Override
+    public void onActivityResumed(Activity activity) {}
+
+    @Override
+    public void onActivitySaveInstanceState(Activity activity, Bundle outState) {
+      outState.putBoolean(STATE_RESOLVING_ERROR, GoogleSignInPlugin.this.resolvingError);
+    }
+
+    @Override
+    public void onActivityStarted(Activity activity) {
+      if (!resolvingError
+          && activity == GoogleSignInPlugin.this.activity
+          && googleApiClient != null) {
+        googleApiClient.connect();
+      }
+    }
+
+    @Override
+    public void onActivityStopped(Activity activity) {
+      if (activity == GoogleSignInPlugin.this.activity && googleApiClient != null) {
+        googleApiClient.disconnect();
```

Comment:
Is there any harm in calling `disconnect()` if we're not connected?  If so, then we should check the connection state here.